### DISCUSSION TITLE
How to change Theia root folder?

### DISCUSSION THREAD

**Discussion by charllie** [eclipse-theia/theia#8457] [General] [answered]
Hello,

I'm starting with Theia and I'm looking for a way to change Theia root folder.

I have this resource tree:
```
c:
|-- a
    |-- b
        |--c
        |--d
```

Currently, I can access to d with http://localhost:3000/#/c:/a/b/d
Is it possible to restrict only to b (http://localhost:3000/#/d)

Also, in workspace folder explorer, I have access to the whole disk and so, it is also possible to restrict it to b?

Thanks

**Comment by vince-fugnitto** [comment, marked as answer]
@charllie thank you for the question, in order to perform restrictions or limitations on the filesystem, some modifications through custom extensions are necessary as highlighted in a previous discussion on the same topic: https://github.com/eclipse-theia/theia/issues/3372. 

I also found some additional information:
- https://spectrum.chat/theia/general/theia-can-access-the-whole-os-file-system~e15507b0-96bf-4321-baab-bc475886b80d

**Reply by charllie** [reply]
Thanks!
